src/components/Categories-button/Categories.tsx

```diff
@@ -1,48 +1,58 @@
 import { useEffect, useState } from 'react'
-import { useSelector } from 'react-redux'
-
+import { useSelector, useDispatch } from 'react-redux'
+import { useNavigate, useParams } from 'react-router-dom'
+import { setOffset, setOperation } from '../../redux/products/slice'
+import { RootState, AppDispatch } from '../../redux/store/index'
+import { ProductCategoryModel } from '../../types/index'
 import styles from './Categories.module.scss'
-import { RootState } from '../../redux/store'
 
 const CategoriesButtons = () => {
-  const [activeButton, setActiveButton] = useState<string | null>(null)
+  const navigate = useNavigate()
+  const { category }: any = useParams()
+  const [activeButton, setActiveButton] = useState(parseInt(category) || 'all')
   const allCategories = useSelector(
     (state: RootState) => state.items.allCategories
   )
+  const dispatch = useDispatch<AppDispatch>()
 
   useEffect(() => {
-    if (allCategories && allCategories.length > 0) {
-      setActiveButton(allCategories[0].name)
+    setActiveButton(parseInt(category) || 'all')
+  }, [category])
+
+  const handleClickButton = (categoryId: number | string) => {
+    navigate(`/catalog/${categoryId}/1`)
```

Comment:
what if here `categoryId` would be `'all'`?

Reply:
in this case, the request for goods is sent without any settings and we will receive goods not divided into categories